### PR TITLE
feat: Add `addContracts` to contract invocation scope

### PR DESCRIPTION
**Pull request by QuinnLee**
Following the recommendation from https://github.com/FuelLabs/fuels-ts/issues/386

### CODE REVIEW

File: packages/contract/src/__test__/contract.test.ts [hunk 1]
```diff
@@ -125,6 +154,19 @@ describe('Contract', () => {
...
+    const contract = await setup();
+
+    const scope = contract
+      .multiCall([contract.functions.foo(1336), contract.functions.foo(1336)])
+      .addContracts([contract.id]);
+
+    expect(scope.transactionRequest.getContractInputs()).toEqual([
+      { contractId: contract.id, type: 1 },
+    ]);
+    expect(scope.transactionRequest.getContractOutputs()).toEqual([{ type: 1, inputIndex: 0 }]);
```
Comment (QuinnLee) [inline packages/contract/src/__test__/contract.test.ts:167]: Not sure what the best way to test this was, I ended up doing this.

File: packages/contract/src/contracts/functions/base-invocation-scope.ts [hunk 2]
```diff
@@ -157,6 +158,15 @@ export class BaseInvocationScope<TReturn = any> {
     return this;
   }
 
+  addContracts(contracts: Array<ContractIdLike> | ContractIdLike) {
```
Comment (luizstacio) [inline packages/contract/src/contracts/functions/base-invocation-scope.ts:161]: - I think because of the name it should always receive an array.

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by luizstacio — APPROVED** [review]

File: packages/contract/src/__test__/call-test-contract/src/main.sw [hunk 3]
```diff
@@ -40,6 +42,11 @@ impl TestContract for Contract {
     log(value);
     value + 1
   }
+  fn call_external_foo(param: u64, contract_id: b256) -> u64 {
+    let external_contract = abi(TestContract, contract_id().into());
+    let response = external_contract.foo(param);
+    response + 1
```
Comment (QuinnLee) [inline packages/contract/src/__test__/call-test-contract/src/main.sw:48]: decided to do so we can verify from the test

File: packages/contract/src/__test__/call-test-contract/src/main.sw [hunk 4]
```diff
@@ -40,6 +42,11 @@ impl TestContract for Contract {
     log(value);
     value + 1
   }
+  fn call_external_foo(param: u64, contract_id: b256) -> u64 {
```
Comment (QuinnLee) [inline packages/contract/src/__test__/call-test-contract/src/main.sw:45]: I was getting errors when I set `contract_id: ContractId` - I'll open an issue to investigate

**Review by luizstacio — APPROVED** [review]

**Pull request merged**